### PR TITLE
chore: use url library for constructing URLs

### PR DESCRIPTION
**Pull request by Piccirello**
Closes ENG-5590.

### CODE REVIEW

**Review by Kunamatata — DISMISSED** [review]
Hype!

File: pkg/http/api.go [hunk 1]
```diff
@@ -263,9 +302,14 @@ func GetSecretNames(host string, verifyTLS bool, apiKey string, project string,
...
-	statusCode, _, response, err := GetRequest(host, verifyTLS, apiKeyHeader(apiKey), "/v3/configs/config/secrets/names", params)
+	url, err := generateURL(host, "/v3/configs/config/secrets/names", params)
 	if err != nil {
-		return nil, Error{Err: err, Message: "Unable to fetch secret names", Code: statusCode}
+		return nil, Error{Err: err, Message: "Unable to generate url"}
+	}
+
+	statusCode, _, response, err := GetRequest(url, verifyTLS, apiKeyHeader(apiKey))
+	if err != nil {
+		return nil, Error{Err: err, Message: "Unable to generate url"}
```
Comment (nmanoogian) [inline pkg/http/api.go]: nit: This error message probably shouldn't have changed

**Review by nmanoogian — DISMISSED** [review]

File: pkg/http/api.go (hunk 1 above)
Comment (Piccirello) [reply pkg/http/api.go]: I was fully confident that I messed up at least one of these. Now that this has been caught, I can relax!

**Review by nmanoogian — APPROVED** [review]

File: pkg/http/api.go (hunk 1 above)
Comment (nmanoogian) [reply pkg/http/api.go]: No worries - I went through 'em all too. That's what reviews are for 👍

**Pull request merged**